### PR TITLE
Adds eth2 network

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- https://github.com/obscuronet/obscuro-internal/issues/1359

### What changes were made as part of this PR:

- Provide a high level list of the changes made

### Definition of done

- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated 
- [ ] End-to-end tests run if required (see below)

### End-to-end tests

Running the end-to-end tests in the [obscuro-test repo](https://github.com/obscuronet/obscuro-test) is currently at the 
discretion of the developer prior to merging a PR. The intention is not to slow down development by having the longer 
running end-to-end tests as a PR gate run on each branch commit etc. To manually trigger a run pre-merge;

- Go to [run_local_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_local_tests.yml)
- Click the "Run workflow" button
- Use the main branch of obscuro-test 
- Enter the name of the PR branch for go-obscuro
- Run the workflow

The run takes approximately 20 minutes and any failure will be reported in the workflow output. Should the tests fail 
the docker container output for the local testnet and all test artifacts will be added to the run and can be downloaded
for debugging and analysis, with a retention period of 2 days. 

Note that every PR merge to main will also trigger a run of post-merge tests, so even if you do not manually trigger 
pre-merge, tests will be run post-merge. The intention being to catch any issues fast on main to allow for quick 
resolution. The post-merge test output can be seen at 
[run_merge_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_merge_tests.yml) and will show both 
the PR number and author in the list of runs.

### CODE REVIEW

**Review by StefanIliev545 — CHANGES_REQUESTED** [review]
initial quick review, will look again

File: integration/eth2network/eth2_configs.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+# Time parameters
+SECONDS_PER_SLOT: 6
+SLOTS_PER_EPOCH: 6
+
+# Deposit contract
+DEPOSIT_CONTRACT_ADDRESS: 0x4242424242424242424242424242424242424242
+`
+
+const baseGenesis = `{
```
Comment (BedrockSquirrel) [inline integration/eth2network/eth2_configs.go:45]: Is there a way we can pull out some of these constants so they're accessible from go? 

In particular I was thinking `DefaultChainID` and `PrefundedAddresses` would be useful.

File: integration/eth2network/eth2_network.go [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+		"--http.port", fmt.Sprintf("%d", httpPort),
+		"--http.api", "admin,miner,engine,personal,eth,net,web3,debug",
+		"--ws",
+		"--ws.port", fmt.Sprintf("%d", wsPort),
+		"--authrpc.port", fmt.Sprintf("%d", executionPort),
+		"--port", fmt.Sprintf("%d", networkPort),
+		"--networkid", fmt.Sprintf("%d", n.chainID),
+		"--syncmode", "full",
+		"--allow-insecure-unlock",
+		"--nodiscover",
```
[the remark this answers is not in the document]
Comment (otherview) [reply integration/eth2network/eth2_network.go:314]: Good point, added both.

File: integration/eth2network/eth2_network.go [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+		eg.Go(func() error {
+			return n.waitForNodeUp(nodeID, 30*time.Second)
+		})
+	}
+	err = eg.Wait()
+	if err != nil {
+		return err
+	}
+
+	// link nodes together by providing the enode node 0
```
[the remark this answers is not in the document]
Comment (otherview) [reply integration/eth2network/eth2_network.go:207]: node ethereum address. Tells node 0 the addresses of the remaining nodes. I'll update the comment, can be clearer 👍

File: integration/eth2network/eth2_configs.go (hunk 1 above)
Comment (otherview) [reply integration/eth2network/eth2_configs.go:45]: The DefaultChainID and the PrefundedAddresses are specified when starting the eth2network.
In: 
https://github.com/obscuronet/go-obscuro/pull/1050/files#diff-a71dc2c1219aeb7132c3bc95c0aea57e9431fce002eb39669d2ffb6e8080a03aR52-R61

The config values in the `eth2_configs.go` should never be used. But I see where you're coming from. The configs are there to enable the POS network. I'll add a comment specifying it.

Does that make sense ?

Comment (BedrockSquirrel) [reply integration/eth2network/eth2_configs.go:50]: Oh I see, yeah that make sense 👍

**Review by StefanIliev545 — APPROVED** [review]
LGTM

**Pull request merged**